applications/ParticleMechanicsApplication/custom_conditions/grid_based_conditions/mpm_grid_surface_load_condition_3d.cpp

```diff
@@ -308,9 +308,9 @@ void MPMGridSurfaceLoadCondition3D::CalculateAll(
         array_1d<double, 3> gauss_load = surface_load;
         for (unsigned int ii = 0; ii < number_of_nodes; ++ii)
         {
-            if( rGeom[ii].SolutionStepsDataHas( SURFACE_LOAD ) )
+            if( r_geometry[ii].SolutionStepsDataHas( SURFACE_LOAD ) )
             {
-                noalias(gauss_load) += N[ii]*rGeom[ii].FastGetSolutionStepValue( SURFACE_LOAD );
+                noalias(gauss_load) += N[ii]*r_geometry[ii].FastGetSolutionStepValue( SURFACE_LOAD );
```

Comment:
When you extract N you're using a row function. This is not needed here. Row makes a copy, thus, it is not really efficient.